pi5/pinctrl_utils.go

```diff
@@ -286,8 +289,88 @@ func (b *pinctrlpi5) createGPIOVPage(memPath string) error {
 	return err
 }
 
-// Cleans up mapped memory / files upon board close() call
-func (b *pinctrlpi5) cleanupPinControlMemory() error {
+/*
+For all Pins belonging to the same bank, pin data is stored contiguously and in 8 byte chunks.
+For a given pin, this method determines:
+ 1. which bank the pin belongs to
+ 2. the starting address of its 8 byte data chunk
+*/
+func getPinAddressOffset(pinNumber int) (int64, error) {
+
+	bankHeaderSize := 8 // 8 bytes of either header data assosciated with a bank (unsure about what is stored here though)
+
+	if !(1 <= pinNumber && pinNumber <= maxPinNum) {
+		return -1, errors.New("pin is out of bank range")
+	}
+
+	for i := 0; i < len(bankDivisions)-1; i++ {
+		if bankDivisions[i] <= pinNumber && pinNumber < bankDivisions[i+1] {
+
+			bankNum := i
+			bankBaseAddr := fselBankOffsets[bankNum]
+			pinBankOffset := pinNumber - bankDivisions[i]
+
+			pinAddressOffset := bankBaseAddr + bankHeaderSize + ((pinBankOffset) * fselPinDataSize)
+			return int64(pinAddressOffset), nil
+		}
+	}
+
+	return -1, errors.New("pin in bank range but not set")
+}
+
+// This method updates the given mode of a pin by finding its specific location in memory & writing to the 'mode' byte in the 8 byte block of pin data.
+func (b *pinctrlpi5) setPin(pinNumber int, newMode byte) error {
+
+	// Of the 8 bytes that represent a given pin's data, only the 4th index corresponds to the alternative mode setting
+	altModeIndex := 4
+
+	pinAddressOffset, err := getPinAddressOffset(pinNumber)
+	if err != nil {
+		return fmt.Errorf("error getting gpio bank number: %w", err)
+	}
+
+	// find pin data within virtual page; retrieve the 4th byte from the pin data
+	pinBytes := b.vPage[pinAddressOffset : pinAddressOffset+fselPinDataSize]
+	altModeByte := pinBytes[altModeIndex]
+
+	// We keep the left half of the byte preserved, only modifying the right half
+	// Preserve Left Side of Byte using this Mask
+	leftSideMask := byte(0xf0)
+
+	// Set Right Side of Byte ONLY using this Mask. Ensures left side cannot be overwritten
+	rightSideMask := byte(0x0f)
+
+	// Set new mode via write to correct while protecting previous other settings
+	newAltModeByte := (altModeByte & leftSideMask) | (newMode & rightSideMask)
+	pinBytes[4] = newAltModeByte
+
+	return err
+}
+
+// Sets up GPIO Pin Memory Access by parsing the device tree for relevant address information
+func (b *pinctrlpi5) setupPinControl() error {
+	nodePath, err := b.findPathFromAlias(gpioName) // this ("gpio") is hardcoded now, we will fix that later!
```

Comment:
For things that you intend to fix later, could you start the comment with `TODO`? That's the usual way to indicate that something here needs to be worked on in the future.

Same comment with the work to do in `cleanFilePath()`